libs/ngx-charts-on-fhir/src/lib/fhir-chart/fhir-chart-configuration.service.ts

```diff
@@ -113,6 +113,39 @@ export class FhirChartConfigurationService {
     }
   }
 
+  zoomIn() {
+    if (this.chart) {
+      const range = this.calculateZoomRange(0.1);
+      this.zoom(range);
+    }
+  }
+
+  zoomOut() {
+    if (this.chart) {
+      const range = this.calculateZoomRange(-0.1);
+      this.zoom(range);
+    }
+  }
+
+  private calculateZoomRange(factor: number): { min: number; max: number } {
+    if (this.chart) {
+      const currentBounds = this.chart.scales['x'].getUserBounds();
+      if (!isNaN(currentBounds.min) && !isNaN(currentBounds.max)) {
+        const rangeSpan = currentBounds.max - currentBounds.min;
+        const zoomSpan = rangeSpan * factor;
+        const newMin = currentBounds.min + zoomSpan / 2;
+        const newMax = currentBounds.max - zoomSpan / 2;
+        return { min: newMin, max: newMax };
+      }
+    }
+    return { min: 0, max: 0 };
+  }
+
+  isFormElementFocused(): boolean | null {
+    const activeElement = document.activeElement;
+    return (activeElement && ['INPUT', 'TEXTAREA', 'SELECT', 'BUTTON'].includes(activeElement.tagName)) || activeElement?.closest('.mat-calendar') !== null;
+  }
```

Comment:
Might make more sense to put this method in FhirChartComponent?
The zoom functions make sense here because they use other internal properties of this service, but this service is not responsible for anything related to form elements. That's more of a component-level concern.